python/get_hatchery_data.py

```diff
@@ -52,4 +53,44 @@ def get_fish_data(self, hatchery, date=None) -> Any:
                     OR species='Sockeye'
                 )""",
         )
-        return results
+        return WDFW_to_df(results)
+    
+    def find_distinct_populations(self, df) -> Any:
+        """
+        Parameters:
+            df: pandas dataframe of data returned from getWDFWData
+        Output: 
+            distinct_populations: List of dictionaries
+                example: [{'species': 'Chinook', 'origin': 'WILD', run: 'Summer'}]
+        """
+        # Group by species -> run -> origin
+        species = df.species.unique()
+
+        distinctPopulations = []
+
+        for s in species:
+            df_s = df[df.species == s]
+            run = df_s.run.unique()
+            for r in run:
+                df_r = df_s[df_s.run == r]
+                origin = df_r.origin.unique()
+                for o in origin:
+                    distinctPopulations.append(
+                        {'species': s, 'run': r, 'origin': o,})              
+        return distinctPopulations
+
+def WDFW_to_df(results):
```

Comment:
This could also be private

Reply:
Or keep static. Up to you!

Reply:
yeah low key this probably should just be private. in dev we you can instantiate a WDFWProviderLogic by doing WDFWProviderLogic(), then you can call the private method like 
`
provider = WDFWProviderLogic()
provider._WDFW_to_df()
`